pgmax/fg/graph.py

```diff
@@ -53,48 +54,48 @@ class FactorGraph:
             this input, and the individual VariableGroups will need to be accessed by indexing.
     """
 
-    variables: Union[
-        Mapping[Any, groups.VariableGroup],
-        Sequence[groups.VariableGroup],
-        groups.VariableGroup,
-    ]
+    variables: Union[groups.VariableGroup, Sequence[groups.VariableGroup]]
 
     def __post_init__(self):
-        if isinstance(self.variables, groups.VariableGroup):
-            self._variable_group = self.variables
+        import time
+
+        start = time.time()
+        if isinstance(self.variables, (vgroup.NDVariableArray, vgroup.VariableDict)):
+            self._variable_groups = [self.variables]
```

Comment:
The name `_variable_groups` is confusing after the refactor. We can probably just update `self.variables` (`FactorGraph` is not frozen and we can directly update) so that it's always called variables (and you can get rid of your `else`)